src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -88,13 +91,24 @@ protected OtpUser getUserProfile(Auth0UserProfile profile) {
      * HTTP endpoint to send an SMS text to an {@link OtpUser}'s phone number with a verification code. This is used
      * during user signup (or if a user wishes to change their notification preferences to use a new un-verified phone
      * number).
+     * Before sending a SMS request, this endpoint also saves the submitted phone number, marks it as not verified,
+     * and sets the user's notification channel to 'sms'.
      */
     public VerificationResult sendVerificationText(Request req, Response res) {
         OtpUser otpUser = getEntityForId(req, res);
-        if (otpUser.phoneNumber == null) {
-            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "User must have valid phone number to verify.");
+        // Get phone number from the path param.
+        String phoneNumber = req.params(PHONE_PARAM);
+        if (StringUtils.isBlank(phoneNumber)) {
+            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "A phone number must be provided.");
         }
-        Verification verification = NotificationUtils.sendVerificationText(otpUser.phoneNumber);
+
+        // Update OtpUser before submitting SMS request.
+        otpUser.phoneNumber = phoneNumber;
+        otpUser.isPhoneNumberVerified = false;
+        otpUser.notificationChannel = "sms";
```

Comment:
As mentioned below, I'm not sure if we want to set this automatically (considering the "both" option)?

Reply:
Removed.